pkg/ccl/backupccl/backup_test.go

```diff
@@ -11204,3 +11204,44 @@ func TestRestoreMemoryMonitoringMinWorkerMemory(t *testing.T) {
 	actualFingerprints := sqlDB.QueryStr(t, "SHOW EXPERIMENTAL_FINGERPRINTS FROM TABLE restore.bank")
 	require.Equal(t, expectedFingerprints, actualFingerprints)
 }
+
+// TestBoundBackupExportOnRanges tests that export requests never get sent
+// across range boundaries, even if the ranges are empty.
+func TestBoundBackupExportOnRanges(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+	const numAccounts = 1
+
+	waitForSplits := make(chan struct{})
+	var responseCount atomic.Int32
+	params := base.TestClusterArgs{}
+	knobs := base.TestingKnobs{
+		DistSQL: &execinfra.TestingKnobs{
+			BackupRestoreTestingKnobs: &sql.BackupRestoreTestingKnobs{
+				RunBeforeBackupFlow: func(ctx context.Context) {
+					<-waitForSplits
+				},
+				RunAfterExportingSpanEntry: func(ctx context.Context, response *kvpb.ExportResponse) {
+					responseCount.Add(1)
+				},
+			},
+		},
+	}
+	params.ServerArgs.Knobs = knobs
+	tc, sqlDB, _, cleanupFn := backupRestoreTestSetupWithParams(t, singleNode, numAccounts, InitManualReplication, params)
+	defer cleanupFn()
+
+	rowsPerRange := 2
+	numRanges := 30
+	sqlDB.Exec(t, "SET CLUSTER SETTING bulkio.backup.presplit_request_spans.enabled=true")
+	var jobID int
+	sqlDB.QueryRow(t, "BACKUP DATABASE data INTO 'userfile:///backup' WITH detached").Scan(&jobID)
```

Comment:
Could we simplify some of this by issuing the backup without detached but after we've called waitForTableSplit?  I think that would allow us to remove the waitForSplits channel and the WaitForJobToSucceed bit?